src/lib.rs

```diff
@@ -14,7 +16,16 @@ fn download(url: String, filename: String, max_files: usize, chunk_size: usize)
     tokio::runtime::Builder::new_multi_thread()
         .enable_all()
         .build()?
-        .block_on(async { download_async(url, filename, max_files, chunk_size).await })
+        .block_on(async { download_async(url, filename.clone(), max_files, chunk_size).await })
+        .map_err(|err| {
+            let path = Path::new(&filename);
+            if path.exists() {
+                remove_file(filename)
+                    .map_err(|err| panic!("Error while removing corrupted file: {:?}", err))
```

Comment:
I panic here which is not great... Is there a way to chain exceptions in PyO3? I couldn't find anything.

Reply:
What do you mean chain ? `PyException::new_err(string)` to create a new Python exception.

Reply:
I mean that this exception will happen while handling another exception. In python you would see something like: 

```python
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: first error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ValueError: second error
```

I don't know if we have the same behaviour here.

Reply:
Oh, that I don't know.

Reply:
For warnings I had to manually call the python runtime through rust. Not sure about chained exceptions.

Reply:
https://github.com/huggingface/safetensors/pull/182/files#diff-57af18dbd0070e35a14e183a52ece76c980f7c45f3dd3835cfacfcc0669a9884R492

Reply:
Do you preffer I change to a warning or this is fine?

Reply:
It's ok for now. This package is not stable in any form anyway.